### ISSUE TITLE
FormatException: Invalid radix-10 number (at character 1)

### ISSUE THREAD

**Issue by ChrisElliotUK** [transistorsoft/flutter_background_geolocation#1294]
## Your Environment
* Plugin version:4.15.2
* Platform: Android
* OS version: 
* Device manufacturer / model: Google Pixel 6a
* Flutter info (`flutter doctor`): 
```
[✓] Flutter (Channel stable, 3.19.6, on macOS 14.4.1 23E224 darwin-arm64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.3)
[✓] VS Code (version 1.89.0)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!
```
* Plugin config:
```
 final bg.Config _locationConfig = bg.Config(
      desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
      useSignificantChangesOnly: false,
      pausesLocationUpdatesAutomatically: false,
      disableElasticity: true,
      isMoving: true,
      showsBackgroundLocationIndicator: true,
      allowIdenticalLocations: true,
      activityType: bg.Config.ACTIVITY_TYPE_FITNESS,
      debug: AppLogs.geolocationDebug,
      logLevel: AppLogs.geolocationLogLevel,
      stopOnStationary: false,
      locationTimeout: 120,
      stopDetectionDelay: 20,
      distanceFilter: 10,
      reset: true,
      locationAuthorizationRequest: "Always",
    );

```

```dart <-- Syntax highlighting: DO NOT REMOVE -->
  Future<Either<Failure, bool>> changePace() async {
    try {
      final result = await backgroundGeolocationDataSource.changePace(false);

      return Right(result);
    } catch (e) {
      print(e.toString());
      return Left(GeoLocationFailure(errorMessage: e.toString()));
    }
  }

Future<bool> changePace(bool isMoving) async =>
      await bg.BackgroundGeolocation.changePace(isMoving);
```

## Expected Behavior
Should throw Platform excpetion with background geolocation is disabled. 

## Actual Behavior
has a format exception
```
FormatException: Invalid radix-10 number (at character 1)
BackgroundGeolocation is disabled
 ^
```

## Steps to Reproduce

1. Call changePace when plugin is disabled
2.
3.
4.

## Context

Trying to test what happens if location permission is set to denied 

## Debug logs

<details>
	<summary>Logs</summary>

``` <!-- syntax-highligting:  DO NOT REMOVE -->
D/TSLocationManager( 7650): [c.t.l.a.BackgroundGeolocation removeListener] 
D/TSLocationManager( 7650):   ✅  removeListener event: providerchange
D/TSLocationManager( 7650): [c.t.l.a.BackgroundGeolocation removeListener] 
D/TSLocationManager( 7650):   ✅  removeListener event: motionchange
D/TSLocationManager( 7650): [c.t.l.a.BackgroundGeolocation removeListener] 
D/TSLocationManager( 7650):   ✅  removeListener event: location
D/TSLocationManager( 7650): [c.t.l.adapter.TSConfig e] ℹ️   Persist config, dirty: [isMoving]
I/flutter ( 7650): FormatException: Invalid radix-10 number (at character 1)
I/flutter ( 7650): BackgroundGeolocation is disabled
I/flutter ( 7650): ^
```

</details>

**Comment by christocracy** [comment]
It’s invalid to call .changePace when the plug-in is disabled.

you should be catching your exceptions

**Comment by ChrisElliotUK** [comment]
@christocracy I am catching it, I am just letting you know the exception that is thrown is not formatted correctly. I fully expect the error and I am just testing different scenarios. My understanding is that it should be a Platform Exception but we get FormatException

**Comment by christocracy** [comment]
I’ll look into it.

**Comment by christocracy** [comment]
I found the problem.  The fix will arrive in the next version.

**Comment by christocracy** [comment]
Released to `4.15.3`.
